### ISSUE TITLE
CoskewACX has some gaps

### ISSUE THREAD

**Issue by aneuhierl** [OpenSourceAP/CrossSection#91]
CoskewACX has some periods with zero valid observations.  Below is some code to reproduce the issue.

```
# Andrew 2020 04

# creates what you need to do machine learning style stuff
# downloads downloadable wide signals
# downloads crsp predictors
# merges together and outputs

# takes about 5 minutes

# creates:
#   temp/signed_predictors_all_wide.csv
#   temp/SignalDocumentation.xlsx



# ==== ENVIRONMENT ====
rm(list = ls())

library(tidyverse)
library(googledrive)
library(data.table)
library(fst)
library(getPass)
library(RPostgres)

# root of April 2021 release on Gdrive
# pathRelease = 'https://drive.google.com/drive/folders/1I6nMmo8k_zGCcp9tUvmMedKTAkb9734R'

# root of March 2022 release on Gdrive
pathRelease = 'https://drive.google.com/drive/u/0/folders/1O18scg9iBTiBaDiQFhoGxdn4FdsbMqGo'


# login to gdrive
# this prompts a login
pathRelease %>% drive_ls()

dir.create('temp/')

# download
target_dribble = pathRelease %>% drive_ls() %>% 
  filter(name == 'Firm Level Characteristics') %>% drive_ls() %>% 
  filter(name == 'Full Sets') %>% drive_ls() %>% 
  filter(name == 'signed_predictors_dl_wide.zip') 
dl = drive_download(target_dribble, path = 'temp/deleteme.zip', overwrite = T)

# unzip, read, clean up
unzip('temp/deleteme.zip', exdir = 'temp')
wide_dl_raw = fread('temp/signed_predictors_dl_wide.csv')
file.remove('temp/signed_predictors_dl_wide.csv')

# check
rel_chars <- "CoskewACX"
count_dt <- wide_dl_raw[, lapply(.SD,function(x) sum(!is.na(x))), .SDcols=rel_chars,  by=list(yyyymm)]

# which months have no obs
count_dt[CoskewACX==0 & yyyymm>=196301 & yyyymm<202201,][order(yyyymm)]

```

**Comment by chenandrewy** [comment]
Thanks @aneuhierl for the reproducible code!   Here is the output for ease of reference 

![image](https://user-images.githubusercontent.com/65506481/173713494-a61dba52-1fed-472c-93aa-372bd9a3227e.png)

(CoskewACX is the # of observations in the month)

And here is a bit more detail of the time-series:
![image](https://user-images.githubusercontent.com/65506481/173713580-57db5266-1519-4889-ae27-0f14fa33c867.png)

![image](https://user-images.githubusercontent.com/65506481/173713597-68653c54-bb92-4739-8bf0-f14b46f6d16d.png)

![image](https://user-images.githubusercontent.com/65506481/173713634-9f290f97-6837-499b-a635-6adf69cb2261.png)

These are strange patterns.  It might be due to line 67 in [CoskewACX.do](https://github.com/OpenSourceAP/CrossSection/blob/master/Signals/Code/Predictors/CoskewACX.do):

```
* exclude of more than five missing obs (just above eq B-7)	
drop if nobs <= 252-5  
```

We'll look into it but it might take some time since this code does rolling estimates using daily data.

**Comment by chenandrewy** [comment]
There also are gaps, in-sample, for the LS portfolios for ChNAalyst, CoskewACX, and iomom_supp

![image](https://user-images.githubusercontent.com/65506481/216102335-81ef36f9-a5c5-43b7-baa0-46e02aef0d57.png)

**Comment by chenandrewy** [comment]
For CoskewACX, `drop if nobs <= 252-5 ` seems to be the problem.  1968 June is one of the bad months, and the 12 months the end of June 1968 had only 246 trading days.

**Comment by chenandrewy** [comment]
CoskewACX is fixed here: f5a618356ba54666127408341a6dba69d4d440bf.  I made new issues for the other gaps.
